### PR TITLE
Move android poc branch in android plugin

### PR DESCRIPTION
**Pull request by jguerra**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,2 +1,4 @@
 version=1.6.6-SNAPSHOT
 groupId=com.pandora.hydra
+sonatypeUsername=CHANGEME
```
Comment (alex-dubrouski) [inline gradle.properties]: Please remove it, it should be in your user's gradle.properties

File: hydra-android-gradle-plugin/src/main/resources/META-INF/gradle-plugins/com.pandora.hydra.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+implementation-class=com.pandora.hydra.HydraAndroidPlugin
```
Comment (alex-dubrouski) [inline hydra-android-gradle-plugin/src/main/resources/META-INF/gradle-plugins/com.pandora.hydra.properties]: Need to rename meta property file, so core and Android plugin will have different IDs

File: hydra-android-gradle-plugin/src/main/java/com/pandora/hydra/android/HydraAndroidPlugin.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        if (project.getSubprojects().isEmpty()) {
+            project.getLogger().info("Applying to leaf project: " + project.getName());
+            HydraPluginExtension hydraExtension = project.getExtensions().create("hydra", HydraPluginExtension.class);
+            if(hydraExtension.isBalanceThreads()) {
+                project.getLogger().info("Hydra Android doesn't support thread balancing. Ignoring setting");
+                hydraExtension.setBalanceThreads(false);
+            }
+            project.afterEvaluate(p -> BalancedTestFactory.createBalancedTest(p, hydraExtension, AndroidUnitTest.class));
+        } else {
+            project.getLogger().lifecycle("Hydra Android mod should not be applied on entire tree, please apply it at app module level");
```
Comment (alex-dubrouski) [inline hydra-android-gradle-plugin/src/main/java/com/pandora/hydra/android/HydraAndroidPlugin.java]: Probably need to clarify/fix this message. Overall meaning is that android plugin should be applied to Android application leaf projects only, as they declare Android unit tests, though there might be Android library leaf projects which probably should use core plugin. Android plugin should never be applied to entire tree of Gradle projects.

File: hydra-android-gradle-plugin/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+       reportOn integrationTest
+   }
+   
+   task allReport(dependsOn: ['unitTestReport', 'integrationTestReport', 'testReport'])
+   
+   test {
+       maxParallelForks = 8
+   }
+   
+   hydra {
```
Comment (alex-dubrouski) [inline hydra-android-gradle-plugin/README.md]: Documentation should be updated as android plugin will have different ID and there will be no explicit task definitions as Android plugin generates them on the fly for all flavors. Thus this file should be completely unique

File: hydra-gradle-plugin/src/main/resources/META-INF/gradle-plugins/com.pandora.hydra.properties [hunk 5]
```diff
@@ -1 +1 @@
-implementation-class=com.pandora.hydra.HydraPlugin
\ No newline at end of file
+implementation-class=HydraPlugin
```
Comment (alex-dubrouski) [inline hydra-gradle-plugin/src/main/resources/META-INF/gradle-plugins/com.pandora.hydra.properties]: It should be fully qualified name, plus we need to make sure these meta files are unique for two plugins

**Comment by jguerra** [comment]
Documentation changes look good. Thanks!

**Pull request merged**